### PR TITLE
[Model Update] BOM As Specified 

### LINKED ISSUE #120: [Model Update]:
## Update Reason
We'd like to update the existing BomAsSpecified to add more usability / value to the model. As for now, the model is attached as a submodel to a digital twin without any reference point to the parts that are mentioned in the BOM. 
In order to change that, we would like to add the propoerty "childCatenaXId" in order to reference to other Parts (e.g. PartAsSpecified).
By doing so, the Item Relationship Service can be integrated and use the parent-child relationship to depict a complete vehicle BOM.

## MS1 Criteria

- [ ] The model that should be updated exists
- [ ] The referenced use case exists
- [ ] The potential updates are discussed with all stakeholders
- [ ] The potential update will be either 
  - [ ] backward compatible or 
  - [ ] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed

### PR DESCRIPTION
**Pull request by maxim-bo**
## Description

 -->

Closes #120 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the BAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar bamm-cli.jar -i \<path-to-aspect-model\> -v ). The  BAMM CLI is available [here](https://openmanufacturingplatform.github.io/sds-documentation/sds-developer-guide/dev-snapshot/tooling-guide/bamm-cli.html) and in [GitHub](https://github.com/OpenManufacturingPlatform/sds-sdk/releases)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "name" and "description"** in English language. 
- [x] **no duplicate names or preferredNames** within an Aspect (e.g. a Property and the referenced Characteristic should not have the same name)
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the BAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://openmanufacturingplatform.github.io/sds-documentation/bamm-specification/v1.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'

### CODE REVIEW

File: io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl [hunk 1]
```diff
@@ -0,0 +1,173 @@
+#######################################################################
+# Copyright (c) 2022 T-Systems International GmbH
+# Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+# Copyright (c) 2022 Volkswagen AG
+# Copyright (c) 2022 Robert Bosch GmbH
+# Copyright (c) 2022 ZF Friedrichshafen AG
+# Copyright (c) 2022 Contributors to the Eclipse Foundation
+#
```
Comment (bs-jokri) [inline io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl:8]: are there any adjustments required to the list of copyright holders?

File: io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    bamm:preferredName "Child Parts Characteristic"@en;
+    bamm:description "The characteristic of the child part property"@en;
+    bamm:dataType :ChildPartsEntity.
+:lifecycleContext a bamm:Property;
+    bamm:preferredName "Lifecycle Context"@en;
+    bamm:description "The lifecycle phase the item belongs to"@en;
+    bamm:characteristic :LifecycleContextCharacteristic;
+    bamm:exampleValue "asSpecified".
+:LifecycleContextCharacteristic a bamm:Characteristic;
+    bamm:preferredName "Lifecycle Context Characteristic"@en;
```
Comment (bs-jokri) [inline io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl:51]: description and example value missing

File: io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    bamm:properties (:measurementUnit :quantityNumber).
+:PartClassificationEntity a bamm:Entity;
+    bamm:preferredName "Part Classification Entity"@en;
+    bamm:description "The entity encapsulating the properties of the part classification."@en;
+    bamm:properties (:key :value).
+:measurementUnit a bamm:Property;
+    bamm:preferredName "Measurement Unit"@en;
+    bamm:description "Unit of measurement for the quantity of serialized objects."@en;
+    bamm:characteristic bamm-c:UnitReference;
+    bamm:exampleValue "e.g. piece, kw"^^bamm:curie.
```
Comment (bs-jokri) [inline io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl:144]: remove example value as it is not a primitive type (and it is not a valid curie)

File: io.catenax.bom_as_specified/2.0.0/metadata.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{ "status" : "draft"}
```
Comment (bs-jokri) [inline io.catenax.bom_as_specified/2.0.0/metadata.json:1]: needs to be changed to release prior release

**Review by bs-jokri — CHANGES_REQUESTED** [review]
check my comments

File: io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl (hunk 1 above)
Comment (maxim-bo) [reply io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl:8]: We will check this asap and update the list if needed

File: io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl (hunk 2 above)
Comment (maxim-bo) [reply io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl:59]: done

File: io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  bamm:optional "true"^^xsd:boolean
+] [
+  bamm:property :partDescription;
+  bamm:optional "true"^^xsd:boolean
+] [
+  bamm:property :partClassification;
+  bamm:optional "true"^^xsd:boolean
+] :childCatenaXId :lastModifiedOn :createdOn).
+:ownerPartId a bamm:Property;
+    bamm:preferredName "Owner PartId"@en;
```
[the remark this answers is not in the document]
Comment (maxim-bo) [reply io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl]: done

File: io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl (hunk 1 above)
Comment (maxim-bo) [reply io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl:8]: we checked this and updated as needed

File: io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    bamm:description "This is the key field of the component which usually keeps the part numbers used in after-sales, e.g. when repairing broken parts and searching for a replacement. This ownerPartId itself isn't usually bound to one part version, with the assumption that all part versions with the same ownerPartId are mutually interchangeable."@en;
+    bamm:characteristic :CatenaXIdTrait;
+    bamm:exampleValue "uuid!".
+:partVersion a bamm:Property;
+    bamm:preferredName "Part Version"@en;
+    bamm:description "This is the version of the part. The engineering will at times supercede an older part version by a newer one, which might have different material aspects, physical dimensions etc., still maintaining compatibility."@en;
+    bamm:characteristic bamm-c:Text;
```
[the remark this answers is not in the document]
Comment (bs-jokri) [reply io.catenax.bom_as_specified/2.0.0/SingleLevelBomAsSpecified.ttl:99]: understood

**Review by bs-jokri — APPROVED** [review]
LGTM

**Comment by bs-jokri** [comment]
from discussion: Property "LifecycleContext" might not be required. Further discussion needed. @maxim-bo @BenediktMuellerIML @catroest will discuss how to proceed.

**Comment by maxim-bo** [comment]
Modeling_Team

**Pull request merged**